File: 0038-count-and-say/0038-count-and-say.cpp
class Solution {
public:
    string countAndSay(int n) {
        string ans = "1";
        while (--n) {
            string tmp;
            int cnt = 1, n = ans.size();
            char ptr = ans[0];
            
            for (int i = 1; i < n; ++i) {
                if (ptr == ans[i]) {
                    cnt++;
                } else {
                    tmp += to_string(cnt);
                    tmp += ptr;
                    cnt = 1, ptr = ans[i];
                }
            }
            
            tmp += to_string(cnt);
            tmp += ptr;
            
            ans = tmp;
        }
        
        return ans;
    }
};

File: 0038-count-and-say/0038-count-and-say.py
class Solution:
    def countAndSay(self, n: int) -> str:
        ans = '1'
        
        for i in range(n - 1):
            cnt, char = 1, ans[0]
            tmp = []
            
            for j in range(1, len(ans)):
                if char == ans[j]:
                    cnt += 1
                else:
                    tmp += str(cnt)
                    tmp += char
                    cnt, char = 1, ans[j]
            
            tmp += str(cnt)
            tmp += char
            
            ans = tmp
                    
        return ''.join(ans)